### PR TITLE
{tools}[GCCcore/10.2.0] awscli 1.19.53, 2.1.37

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    ('Python', '3.8.6'),
+    ('PyYAML', '5.3.1'),
+]
+
+use_pip = True
+
+exts_default_options = {'source_urls': [PYPI_SOURCE]}
+
+exts_list = [
+    ('colorama', '0.4.3', {
```
Comment (Micket) [inline easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb:26]: Part of Python 3.8.6 already

File: easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        'checksums': ['e96da0d330793e2cb9485e9ddfd918d456036c7149416295932478192f4436a1'],
+    }),
+    ('s3transfer', '0.3.3', {
+        'checksums': ['921a37e2aefc64145e7b73d50c71bb4f26f46e4c9f414dc648c6245ff92cf7db'],
+    }),
+    ('rsa', '4.4.1', {
+        'checksums': ['efaf0c32afee1c136e5cd2e7ceecf2dfc65dac00fb812a1b3b8b72f6fea38dbb'],
+    }),
+    ('docutils', '0.15', {
```
Comment (Micket) [inline easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb:35]: Version 0.16 is part of Python 3.8.6 already. Do you need to intentionally downgrade it?

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+exts_default_options = {'source_urls': [PYPI_SOURCE]}
+
+exts_list = [
+    ('jmespath', '0.10.0', {
+        'checksums': ['b85d0567b8666149a93172712e68920734333c0ce7e89b78b3e987f71e5ed4f9'],
+    }),
+    ('botocore', '1.20.53', {
+        'source_tmpl': '2c5e5782c4.tar.gz',
+        'source_urls': ['https://github.com/boto/botocore/archive/'],
```
Comment (Micket) [inline easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:30]: Does this need to be different from the other easyconfig?

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        'checksums': ['a8d6340ded75ce69bc10db9c5b37d0ef443fbad417a366c6a702b4a3d4594ad4'],
+    }),
+    ('docutils', '0.15.2', {
```
Comment (Micket) [inline easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:33]: again, 0.16 already part of Python 3.8.6?

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    ('s3transfer', '0.3.7', {
+        'checksums': ['35627b86af8ff97e7ac27975fe0a98a312814b46c6333d8a6b889627bcd80994'],
+    }),
+    ('ruamel.yaml', '0.15.100', {
+        'checksums': ['8e42f3067a59e819935a2926e247170ed93c8f0b2ab64526f888e026854db2e4'],
+    }),
+    ('prompt_toolkit', '2.0.10', {
+        'checksums': ['f15af68f66e664eaa559d4ac8a928111eebd5feda0c11738b5998045224829db'],
+    }),
+    ('cryptography', '3.3.2', {
```
Comment (Micket) [inline easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:45]: 3.1.1 is already included in Python 3.8.6, do we need to update it here?

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        'checksums': ['5a60d3780149e13b7a6ff7ad6526b38846354d11a15e21068e57073e29e19bed'],
+    }),
+    ('distro', '1.5.0', {
+        'checksums': ['0e58756ae38fbd8fc3020d54badb8eae17c5b9dcbed388b17bb55b8a5928df92'],
+    }),
+    ('wcwidth', '0.1.9', {
```
Comment (Micket) [inline easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:51]: 0.2.5 is already included in Python 3.8.6, do you need to downgrade it?

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb (hunk 2 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb:35]: Yes, setup.py has a lot of ==

File: easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb (hunk 1 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-1.19.53-GCCcore-10.2.0-Python-3.8.6.eb:26]: will remove it

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb (hunk 3 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:30]: different? swapping the order of tmpl and URLs?  not sure why I did that

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb (hunk 4 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:33]: setup.py is very strict

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb (hunk 3 above)
Comment (Micket) [reply easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:30]: No, compare with botocore in awscli 1.19.53

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb (hunk 6 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:51]:
```
requires = [
    'botocore==2.0.0dev105',
    'colorama>=0.2.5,<0.4.4',
    'docutils>=0.10,<0.16',
    'cryptography>=3.3.2,<3.4.0',
    's3transfer>=0.3.0,<0.4.0',
    'ruamel.yaml>=0.15.0,<0.16.0',
    # wcwidth 0.2.0 and up do not get copied into
    # the pyinstaller executable.
    'wcwidth<0.2.0',
    'prompt-toolkit>=2.0.0,<3.0.0',
    'distro>=1.5.0,<1.6.0',
]
```

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb (hunk 5 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:45]: ``` 'cryptography>=3.3.2,<3.4.0', ```

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb (hunk 3 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:29]: Yes, botocore does not support AWScli 2 yet. Needs to be pulled from the archive.

File: easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb (hunk 6 above)
Comment (fizwit) [reply easybuild/easyconfigs/a/awscli/awscli-2.1.37-GCCcore-10.2.0.eb:51]: Yes, awscli has very specific requirements.  In cases like this should a patch be made to change setup.py to make versions more liberal?

**Comment by fizwit** [comment]
switch to awscliv2

**Pull request closed without merging**